public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx

```diff
@@ -45,6 +45,17 @@ import { ShareModal } from '../ShareModal';
 
 interface Props extends ShareModalTabProps {}
 
+type SharePublicDashboardAcknowledgmentInputs = {
+  publicAcknowledgment: boolean;
+  dataSourcesAcknowledgment: boolean;
+  usageAcknowledgment: boolean;
+};
+
+export type SharePublicDashboardInputs = {
+  isAnnotationsEnabled: boolean;
+  enabledSwitch: boolean;
+} & SharePublicDashboardAcknowledgmentInputs;
```

Comment:
whats the `& SharePublicDashboardAcknowledgmentInputs` do?

Reply:
neat, learned something new. I knew there were union types, guess it makes sense that the opposite exists

Reply:
that's correct